test/scripts/test_dfu.py

```diff
@@ -0,0 +1,105 @@
+from serial_helper import SerialHelper
+import os
+import struct
+from collections import namedtuple
+from typing import Dict, Any
+import crcmod
+from pathlib import Path
+from func_timeout import func_timeout, FunctionTimedOut
+
+
+class TestDFU:
+
+    SUCCESSFUL_TIMEOUT_S = 60
+
+    def __get_version_from_bin(self, file: str, offset: int = 0) -> Dict[str, Any]:
+        VersionHeader = namedtuple("VersionHeader", "magic gitSHA maj min")
+        magic = 0xDF7F9AFDEC06627C
+        header_format = "QIBB"
+        header_len = struct.calcsize(header_format)
+        version = None
+        with open(file, "rb") as f:
+            data = f.read()
+
+            for offset in range(offset, len(data) - header_len):
+                header = VersionHeader._make(
+                    struct.unpack_from(header_format, data, offset=offset)
+                )
+
+                if magic == header.magic:
+                    version = {
+                        "sha": f"{header.gitSHA:08X}",
+                        "version": f"{header.maj}.{header.min}",
+                    }
+                    break
+        return version
+
+    def __wait_until_update_success(self, ser: SerialHelper) -> bool:
+        ret = False
+        while True:
+            read = ser.read_line()
+            if "Update successful" in read:
+                ret = True
+                break
+            if "Update failed" in read:
+                break
+        return ret
+
+    def test_dfu_perform(self, ser: SerialHelper, file: str):
+        if file is not None:
+            # try:
+            file = os.path.abspath(file)
+            size = os.path.getsize(file)
+            data = Path(file).read_bytes()
+
+            kermit = crcmod.predefined.mkCrcFun("kermit")
+            crc = kermit(data)
+            fw_ver = self.__get_version_from_bin(file)
+            major = int(fw_ver["version"].split(".")[0])
+            minor = int(fw_ver["version"].split(".")[1])
+            sha = int(fw_ver["sha"], 16)
+            tx = (
+                "bm dfu start 0x8b718c26aea126dc "
```

Comment:
Need to add a node_id to the pytest parser, currently the node ID is hardcoded :)